rules/external_binaries/def.bzl

```diff
@@ -12,32 +12,52 @@ def _external_binary_impl(ctx):
         "url": url,
         "sha256": ctx.attr.sha256[os],
     }
+
+    if ctx.attr.strip_prefix.get(os, "") != "":
+        args["stripPrefix"] = ctx.attr.strip_prefix[os]
+
     if any([url.endswith(suffix) for suffix in [".zip", ".tar.gz", ".tgz", ".tar.bz2", ".tar.xz"]]):
-        ctx.download_and_extract(output="{name}/{name}_out".format(name = ctx.attr.name), **args)
+        ctx.download_and_extract(output = ".", **args)
+        build_contents = """
```

Comment:
I'm a bit confused.. is this a generated file? wouldn't make it sense to carry out in in the codebase instead to be in stubbed in the function?

Reply:
A Bazel repository rule generates another Bazel repository that we can load with `@<repository_name>`. So, for the downloaded files, we generate a `BUILD.bazel` that creates a symlink and exports those real files.

Reply:
When we use `@helm//:binary`, it's using the symlink. When we use `@helm//:helm` it's using the real path.

For using the real path, it's less predictable, this is why we create the `:binary` target, for easy discoverability of the binary from the downloaded files. It's useful when consuming `tgz` files that we don't know the exact directory structure.